### PR TITLE
Store address input to local storage

### LINKED ISSUE #34: 📍 Location Caching
**Type**: client
**FOF**:  3

## Description

Please reference #38 to figure out how this works.

Add a context and provider that the application can hook into that stores the address for the application. When a user enters their address on the landing page, it saves this in local storage and the context and redirects to the home page. 

Other pages should be able to hook into this information and display/access it as necessary. 

## Screenshot

Voter FAQ

## Todo

- [ ] Make Provider, Context, and hook
- [ ] Update the hook in the landing page 
- [ ]  Display information in navbar

## Links
- [Figma](https://www.figma.com/file/6aJ3cfu2Yq5dxUQcNDkm6Y/Brink-Designs?node-id=90%3A119)
- #38

### LINKED ISSUE #39: 🗄️ Addresss React Context and Local Storage
**Type**: client
**FOF**:  3

## Description

Please reference #38 to figure out how this works.

Add a context and provider that the application can hook into that stores the address for the application. When a user enters their address on the landing page, it saves this in local storage and the context and redirects to the home page. 

Other pages should be able to hook into this information and display/access it as necessary. 

## Screenshot

Voter FAQ

## Todo

- [ ] Make Provider, Context, and hook
- [ ] Update the hook in the landing page 
- [ ]  Display information in navbar

## Links
- [Figma](https://www.figma.com/file/6aJ3cfu2Yq5dxUQcNDkm6Y/Brink-Designs?node-id=90%3A119)
- #38

### PR DESCRIPTION
**Pull request by suewee**
## Summary

Modify Search Bar to pass down inputs and hook into these values to display on the Search page accordingly.

Closes #34 

## Changes

- Create Address Context and Provider logic
- Store address input in Local Storage
- Redirect user to Landing Page if Address Context is not populated
- Edits to Navbar and cleaned up snippets of code

### Updated Navbar Rendering 
Top is Figma, bottom is localhost: 

image

### CODE REVIEW

File: client/src/components/ui/Autocomplete.tsx [hunk 1]
```diff
@@ -38,12 +43,21 @@ export default function SearchAutocomplete() {
...
+    navigate(createPath(APP_ROUTE.HOME, TAB_LINKS.FAQ)); //go home
+  };
+
   return (
     <AutoComplete
       rollNavigation
       closeOnBlur={false}
       shouldRenderSuggestions={() => ready}
-      onSelectOption={({ item }) => setValue(item.value)}
+      onSelectOption={({ item }) => onAddressSubmit(item.value)}
```
Comment (ayanmallik) [inline client/src/components/ui/Autocomplete.tsx:60]: Great job integrating the onAddressSubmit function into Arman's autocomplete component!

File: client/src/contexts/AddressContext.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const AddressContext = createContext<AddressInterface | undefined>(undefined);
+
+const AddressProvider: FC = ({ children }): ReactElement => {
+  const [address, setAddress] = useState(
+    localStorage.getItem(ADDRESS_KEY) || '',
+  );
+
+  useEffect(() => {
+    console.log('useEffect');
```
Comment (ayanmallik) [inline client/src/contexts/AddressContext.tsx:25]: Can we clear up this console.log lol?

File: client/src/layout/Navbar.tsx [hunk 3]
```diff
@@ -1,32 +1,46 @@
...
+
+  return (
+    <nav className="navbar-wrapper">
+      <ul className="navbar-list">
+        <li className="navbar-item">
+          <Link href="/">
+            <Image src={Logo} alt="Brink" height="30px" />
+          </Link>
+        </li>
+        {address !== (undefined || '') && location.pathname !== '/landing' && (
```
Comment (ayanmallik) [inline client/src/layout/Navbar.tsx:24]: Can we abstract this conditional logic into a function (could call into showAddress())?

**Review by ayanmallik — APPROVED** [review]
Great job making the landing page come to life! The address storage/redirection functionality works great and is super cool

File: client/src/layout/Navbar.tsx (hunk 3 above)
Comment (amit-sawhney) [reply client/src/layout/Navbar.tsx:24]: also make the function follow boolean standards of being some logical word to start such as `shouldShowAddress()`

**Review by amit-sawhney — APPROVED** [review]
Make sure to address the redirecting part :) Looks really good overall!

**Pull request merged**